src/lug/Graphics/Vulkan/Gui.cpp

```diff
@@ -0,0 +1,876 @@
+            if (!imageBuilder.build(_fontImage, &result)) {
+                LUG_LOG.error("Gui::initFontsTexture: Can't create depth buffer image: {}", result);
+                return false;
+            }
+            result = VK_SUCCESS;
+            if (!deviceMemoryBuilder.addImage(_fontImage)) {
+                LUG_LOG.error("Gui::initFontsTexture: Can't add image to device memory");
+                return false;
+            }
+
+            result = VK_SUCCESS;
+            if (!deviceMemoryBuilder.build(_fontDeviceMemory, &result)) {
+                LUG_LOG.error("Gui::initFontsTexture: Can't create staging buffer device memory: {}", result);
+                return false;
+            }
+        }
+    }
+
+    // Create FontsTexture image view
+    {
+        API::Builder::ImageView imageViewBuilder(device, _fontImage);
+
+        imageViewBuilder.setFormat(_fontImage.getFormat());
+        imageViewBuilder.setAspectFlags(VK_IMAGE_ASPECT_COLOR_BIT);
+
+        VkResult result{VK_SUCCESS};
+        if (!imageViewBuilder.build(_fontImageView, &result)) {
+            LUG_LOG.error("Gui::initFontsTexture: Can't create image view: {}", result);
+            return false;
+        }
+    }
+
+    // Create staging buffers for font data upload
+    {
+        API::Buffer stagingBuffer;
+        API::DeviceMemory stagingBufferMemory;
+        std::set<uint32_t> queueFamilyIndices = { _transferQueue->getQueueFamily()->getIdx() };
+
+        // Create staging buffer
+        {
+            API::Builder::Buffer bufferBuilder(device);
+            bufferBuilder.setQueueFamilyIndices(queueFamilyIndices);
+            bufferBuilder.setSize(uploadSize);
+            bufferBuilder.setUsage(VK_BUFFER_USAGE_TRANSFER_SRC_BIT);
+            bufferBuilder.setExclusive(VK_SHARING_MODE_EXCLUSIVE);
+
+            VkResult result{VK_SUCCESS};
+            if (!bufferBuilder.build(stagingBuffer, &result)) {
+                LUG_LOG.error("Gui::initFontsTexture: Can't create staging buffer: {}", result);
+                return false;
+            }
+        }
+
+        // Create staging buffer memory
+        {
+            API::Builder::DeviceMemory deviceMemoryBuilder(device);
+            deviceMemoryBuilder.setMemoryFlags(VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT);
+            deviceMemoryBuilder.addBuffer(stagingBuffer);
+
+            VkResult result{VK_SUCCESS};
+            if (!deviceMemoryBuilder.build(stagingBufferMemory, &result)) {
+                LUG_LOG.error("Gui::initFontsTexture: Can't create staging buffer device memory: {}", result);
+                return false;
+            }
+        }
+
+
+        stagingBuffer.updateData(fontData, (uint32_t)uploadSize);
+
+        // Copy buffer data to font image
+        {
+            VkResult result{VK_SUCCESS};
+            API::CommandBuffer commandBuffer;
+
+            API::Builder::CommandBuffer commandBufferBuilder(_renderer.getDevice(), _commandPool);
```

Comment:
You can't use `_commandPool` if the transfer queue family and the graphic queue family are not the same